spaceship/lib/spaceship/two_step_or_factor_client.rb

```diff
@@ -4,7 +4,7 @@
 module Spaceship
   class Client
     def handle_two_step_or_factor(response)
-      raise "2FA can only be performed in interactive mode" if ENV["FASTLANE_IS_INTERACTIVE"] == "false"
+      raise "2FA can only be performed in interactive mode" if ENV["SPACESHIP_ONLY_ALLOW_INTERACTIVE_2FA"] == "true" && ENV["FASTLANE_IS_INTERACTIVE"] == "false"
```

Comment:
Sorry to get back to a merged PR but I thinks there's a condition which is not quite right:

What happens if you have `FASTLANE_IS_INTERACTIVE=false` so you should skip inmediately this assertion but you have defined this new `SPACESHIP_ONLY_ALLOW_INTERACTIVE_2FA=false` ? You are running in non-interactive but it will raise the non-interactive error.

I think the condition evaluation should be swapped so `SPACESHIP_ONLY_ALLOW_INTERACTIVE_2FA` value makes no effect if running interactive.

Reply:
I don't understand the concern. The only way this exception is thrown is if both ENV variables are set to the `true` and `false` respectively. If you don't set either it will go over the exception and execute as if it's interactive.

Reply:
`FASTLANE_IS_INTERACTIVE` is actually set by _fastlane_ and not the user. It is set in the `Fastlane::CLIToolsDistributor`. The `Spaceship` module does not have access to fastlane stuff so this is used as a way for them to talk when `fastlane spaceauth` is used.

So this gets set automatically by _fastlane_ when in a non-interactive state

Reply:
My bad guys, I don't know what I was thinking. You are totally right.

Reply:
Np np! It's not the most straight forward thing 😇 I appreciate you bringing up your concerns though ❤️